docs/embedding.md

```diff
@@ -0,0 +1,106 @@
+
+That's where Volumes come in - they allow us to have our target data right beside our GPU and load it into our containers concurrently. The feature itself is built to allow for all of your containers to read in data quickly and efficiently at the same time. This is ideal for jobs where we might need to run a large batch job involving a distributed dataset or a model whose weights we'd like to load in quickly. 
+
+It's suprisingly simple to set-up, all you need to create a new volume is the following 5 lines of code
+
+```python
+import modal
+
+stub = modal.Stub()
+stub.volume = modal.volume.new() # Declare a new volume
+
+@stub.function(volumes={"/root/foo": stub.volume})
+def function():
+  // Perform tasks here
+```
+
+If you'd like to ensure that the data in the volume is permanently stored on Modal's servers and can be accessed by any of your other modal functions, all it takes is just swapping out the `.new` for the `.persisted` keyword
+
+```python
+volume = Volume.persisted("Unique Name Here")
+```
+
+Once you have this, you'll be able to access this volume from any other app you define using the `volumes` parameter in the `stub.function` keyword. In fact, what makes this so good is that a file on your volume behaves almost identically to a file located on your local file system, yet mantains all of the advantages that Modal brings. 
+
+In some of our initial test runs, we were able to load in the entirety of English Wikipedia, all 22 GB of it, into a container in just under 5 seconds and start executing jobs. Compare that to the 10 minutes it took to download the dataset from scratch and you can imagine the huge benefits this would bring for any endpoints or batch jobs down the line. 
+
+
+
+# Embedding Wikipedia
+
+Now that we've got a good understanding of some key concepts that Modal provides, let's look at volumes in action by running our first embedding job. We've loaded a small dataset into a directory called `wikipedia` inside a persistent volume we've created called `embedding-wikipedia`
+
+<TODO : Add in a simplified script to run embeddings without the sample portion>
+
+## Scaling Out
+
+Now that we've seen how to run a simple batch job using Modal, let's consider how we might modify our embedding function to bring our current timing of ~4 hours down. 
+
+Well, turns out we have two handy properties that we can adjust
+
+- `concurrency_limit` : This is the number of containers that Modal is allowed to spawn concurrently to run a batch job
+- `allow_concurrent_inputs` : This is the number of inputs a container can handle at any given time
+
+All we really need to do then is to crank up the values of these two properties as seen below and we'll have 50 different containers each with their own A10g GPU processing 40 batches of inputs each at any given time. 
+
+```python
+@stub.cls(
+    gpu=GPU_CONFIG,
+    image=tei_image,
+    concurrency_limit=50, # Number of concurrent containers that can be spawned to handle the task
```

Comment:
Again - trying to highlight how easy it is for someone to get started working with large distributed jobs by adding two config variables and u get a almost 90% reduction in time needed to run the job without even changing the underlying implementation